packages/common/src/api/comments.ts

```diff
@@ -69,22 +85,57 @@ const commentsApi = createApi({
         return commentsRes
       },
       options: { type: 'mutation' },
-      async onQuerySuccess({ data: comment }, { entityId }, { dispatch }) {
-        dispatch(
-          commentsApi.util.updateQueryData(
-            'getCommentsByTrackId',
-            { entityId },
-            (prevState) => {
-              // TODO: how should we handle sorting here?
-              prevState?.unshift(comment) // add new comment to top of comment section
+      async onQuerySuccess(
+        { data: newId },
+        { entityId, body, userId, timestampS, parentCommentId },
+        { dispatch }
+      ) {
+        const newComment: Comment = {
+          id: newId,
+          userId,
+          message: body,
+          isPinned: false,
+          timestampS,
+          reactCount: 0,
+          replies: undefined,
+          createdAt: new Date().toISOString(),
+          updatedAt: undefined
+        }
+        optimisticUpdateCommentList(
+          entityId,
+          (prevState) => {
+            if (prevState) {
+              if (parentCommentId) {
+                const parentCommentIndex = prevState?.findIndex(
+                  (comment) => comment.id === parentCommentId
+                )
+                if (parentCommentIndex && parentCommentIndex >= 0) {
+                  const parentComment = prevState[parentCommentIndex]
+                  parentComment.replies = parentComment.replies || []
+                  parentComment.replies.push(newComment)
+                }
+              } else {
+                prevState.unshift(newComment) // add new comment to top of comment section
+              }
             }
-          )
+          },
+          dispatch
+        )
+        optimisticUpdateComment(
+          parentCommentId ?? newId,
+          (parentComment) => {
+            if (parentCommentId && parentComment) {
+              parentComment.replies = parentComment.replies || []
+              parentComment.replies.push(newComment)
+              return parentComment
+            } else {
+              return newComment
+            }
+          },
+          dispatch
```

Comment:
Quirks of having the individual comments & comment arrays stored separately via audius-query is having to update two SOTs at the same time 😓 (one to make list changes happen, another to make comment metadata changes happen)